test/filters/test_convolution.cpp

```diff
@@ -41,43 +41,59 @@
 #include <pcl/pcl_tests.h>
 #include <pcl/filters/convolution.h>
 #include <pcl/point_types.h>
+#include <cmath>
 
 using namespace pcl;
 using namespace pcl::test;
 
-PointCloud<PointXYZI>::Ptr input (new PointCloud<PointXYZI> ());
-Eigen::ArrayXf filter(7);
+typedef struct
+{
+  float r;
+  float g;
+  float b;
+} COLOR;
+
+static const COLOR basemap[6] = {
+  { 0.2902f, 0.0784f, 0.5490f },
+  { 0.1176f, 0.5333f, 0.8980f },
+  { 0.5451f, 0.7647f, 0.2902f },
+  { 1.0000f, 0.9216f, 0.2314f },
+  { 1.0000f, 0.0000f, 0.0000f },
+};
+
+COLOR get_color(COLOR* colormap, size_t color_count, float fmin, float fmax, float value)
+{
+  float* steps = new float[color_count];
+  float f = (fmax - fmin) / (float)(color_count - 1);
+  for (int i = 0; i < (int)color_count; i++)
+    steps[i] = fmin + f * (float)i;
+  COLOR result = { 0, 0, 0 };
+  if (value < fmin) value = fmin;
+  if (value > fmax) value = fmax;
+  for (size_t i = 1; i < color_count; i++)
+  {
+    if (steps[i - 1] <= value && value <= steps[i])
+    {
+      result.r = (colormap[i - 1].r == colormap[i].r) ? colormap[i].r : (colormap[i - 1].r + ((colormap[i].r - colormap[i - 1].r) / (steps[i] - steps[i - 1])) * (value - steps[i - 1]));
+      result.g = (colormap[i - 1].g == colormap[i].g) ? colormap[i].g : (colormap[i - 1].g + ((colormap[i].g - colormap[i - 1].g) / (steps[i] - steps[i - 1])) * (value - steps[i - 1]));
+      result.b = (colormap[i - 1].b == colormap[i].b) ? colormap[i].b : (colormap[i - 1].b + ((colormap[i].b - colormap[i - 1].b) / (steps[i] - steps[i - 1])) * (value - steps[i - 1]));
```

Comment:
Has `colormap` an iterator, so you could write
```
for (std::size_t c = 0; c < 3; ++c) {
  result.r = (colormap[i - 1][c] == colormap[i][c]) ? colormap[i][c] : (colormap[i - 1][c] + ((colormap[i][c] - colormap[i - 1][c]) / (steps[i] - steps[i - 1])) * (value - steps[i - 1]));
}
```

Reply:
`RGB` has no index operator